smart-contracts/witnet-randomness-oracle/randomness-contract.md

```diff
@@ -0,0 +1,63 @@
+
+Moreover, the same instance can be safely and concurrently used by any number of indepent applications running within the same EVM-chain (i.e. any of those currently supported by the Witnet side-chain). 
+
+  {% hint style="info" %}
+  Because of this, you may even have the chance to reuse randomness previously requested by others _**at no cost**_ (🎉), while having the certainty that the acquired randmoness was independently generated by the Witnet side-chain strictly after any given block number.
+  {% endhint %}
+
+  {% hint style="success" %}
+  And therefore, depending on the time-response constraints of your own app, the Witnet Randomness Oracle becomes the cheapest side-chain mechanism to generate reliable randomness in the whole crypto arena (💣).
+  {% endhint %}
+
+In a nutshell, by using a `WitnetRandomness` instance, you just need to care for requesting and paying for new randomness at any given time (i.e. `randomize() payable`), and eventually fetching the Witnet-provided randomness value (i.e. `isRandomized(uint256)`, `getRandomnessAfter(uint256)`). As simple as that ;)
+
+  {% content-ref url="api-reference.md" %}
+  [api-reference.md](api-reference.md)
+  {% endcontent-ref %}  
+
+
+## Best practices
+
+- The `WitnetRandomness` contract is not owned, so anyone is actually allowed to either request new randomness (i.e. `randomize()`), or get previously solved randomness from storage (i.e. `isRandomized(uint _blockNumber)`, `getRandomnessAfter(uint _blockNumber)`).
+
+- Besides, you or your contract will be able to request for new randomness (i.e. `randomize() payable`), even if other previous randomness requests (either coming from you, or other apps) were still pending to be solved.
+
+- The `WitnetRandomness`contract is "clonable", which means that you, or your contract, may cheaply create your own cloned copy as a means to gain ownership of the self-contained `WitnetRequestRandomness` contract, and therefore be allowed to change the _witnessing parameters_ of the low-level Randomness Request that gets posted to the Witnet side-chain. 
+
+- The `WitnetRandomness`contract provides the `random(uint32 _range, uint256 _nonce, uint256 _block)` and `random(uint32 _range, uint256 _nonce, bytes32 _seed)` helper functions. These functions guarantee that different contracts/apps reading from same randomness request will actually get _completely different random sequences of **uniformly distributed numbers**_, within a given range, and doing so while _keeping complexity under **O(1)**_ (🎉).
+
+- Most of all: the `WitnetRandomness`contract guarantees the returned randomness to be the one requested at the given block number, or the _very next_ randomness request posted after the given block if there was none posted back then, or if the ones posted after that got reverted for whatever reason at the Witnet side-chain level.
```

Comment:
I find this sentence hard to read. Could you split it in different sentences and remove unnecessary words? Consider the following a quick suggestion
```suggestion
- The `WitnetRandomness`contract guarantees that the received randomness is the one requested for the given block number. Or the _subsequent_ randomness request posted after the given block if none were posted, or if the ones posted after that got reverted for whatever reason at the Witnet side-chain level.
```